From 2f74b344d64eb8d01d1ba585f0896a5586ae8754 Mon Sep 17 00:00:00 2001
From: Ramasamy Kaliappan <quic_rkaliapp@quicinc.com>
Date: Mon, 3 Jul 2023 15:54:14 +0530
Subject: [PATCH 04/16] wifi: mac80211: wme: use ap_addr instead of deflink
 BSSID

We use this to look up the destination station, so it
needs to be the MLD address of the AP for an MLO; use
ap_addr instead of the BSSID.

Signed-off-by: Johannes Berg <johannes.berg@intel.com>
---
 net/mac80211/wme.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/mac80211/wme.c b/net/mac80211/wme.c
index b10ca96..b154d05 100644
--- a/net/mac80211/wme.c
+++ b/net/mac80211/wme.c
@@ -214,7 +214,7 @@ u16 ieee80211_select_queue(struct ieee80211_sub_if_data *sdata,
 		if (sta)
 			break;
 
-		ra = sdata->deflink.u.mgd.bssid;
+		ra = sdata->vif.cfg.ap_addr;
 		break;
 	case NL80211_IFTYPE_ADHOC:
 		ra = skb->data;
-- 
2.17.1

